frontend/src/enum/SeverityEnum.ts

```diff
@@ -2,4 +2,15 @@ export enum Severity {
     success = 'success',
     warning = 'warn',
     error = 'error',
-}
\ No newline at end of file
+}
+
+export enum ErrorCode {
+    conflict = 'conflict',
+    selfGrantProhibited = 'selfGrantProhibited',
+    default = 'default',
+}
+
+export enum ErrorDescription{
+    selfGrantProhibited = 'Granting admin privilege to self is not allowed.',
+    default = 'An error has occured.',
+}
```

Comment:
Rarely seen using enum like this `selfGrantProhibited = 'Granting admin privilege to self is not allowed.',`. Often should be constant.